pkg/webhook/common.go

```diff
@@ -0,0 +1,100 @@
+package webhook
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"strings"
+
+	opa "github.com/open-policy-agent/frameworks/constraint/pkg/client"
+	"github.com/open-policy-agent/gatekeeper/apis/config/v1alpha1"
+	"github.com/open-policy-agent/gatekeeper/pkg/controller/config/process"
+	"github.com/open-policy-agent/gatekeeper/pkg/keys"
+	"github.com/open-policy-agent/gatekeeper/pkg/util"
+	authenticationv1 "k8s.io/api/authentication/v1"
+	k8sruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+var log = logf.Log.WithName("webhook")
+
+var (
+	// VwhName is the gatekeeper validating webhook for policy validation
+	VwhName = "gatekeeper-validating-webhook-configuration"
+)
+
+const (
+	serviceAccountName = "gatekeeper-admin"
+)
+
+var (
+	runtimeScheme                      = k8sruntime.NewScheme()
+	codecs                             = serializer.NewCodecFactory(runtimeScheme)
+	deserializer                       = codecs.UniversalDeserializer()
+	disableEnforcementActionValidation = flag.Bool("disable-enforcementaction-validation", false, "disable validation of the enforcementAction field of a constraint")
+	logDenies                          = flag.Bool("log-denies", false, "log detailed info on each deny")
+	emitAdmissionEvents                = flag.Bool("emit-admission-events", false, "(alpha) emit Kubernetes events in gatekeeper namespace for each admission violation")
+	serviceaccount                     = fmt.Sprintf("system:serviceaccount:%s:%s", util.GetNamespace(), serviceAccountName)
+	// webhookName is deprecated, set this on the manifest YAML if needed"
+)
+
+func isGkServiceAccount(user authenticationv1.UserInfo) bool {
+	return user.Username == serviceaccount
+}
+
+type webhookHandler struct {
```

Comment:
nit: opa is validation specific for now

Reply:
Right! Moved opa from the common webhook to the validation webhook.

Reply:
Thanks!